File: ExpenseService/Models/Expense.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ExpenseService.Models
{
    public enum ExpenseCategory
    {
        Prevoz,
        Smestaj,
        Hrana,
        Ulaznice,
        Kupovina,
        Ostalo
    }

    public class Expense
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public Guid TripPlanId { get; set; }

        [Required]
        [MaxLength(150)]
        public string Name { get; set; }

        public ExpenseCategory Category { get; set; } = ExpenseCategory.Ostalo;

        [Required]
        public decimal Amount { get; set; }

        [Required]
        public DateTime Date { get; set; }

        public string Description { get; set; }
    }
}

File: TravelPlannerFrontend/src/models/Expense.js
export const ExpenseCategory = {
  PREVOZ: 'Prevoz',
  SMESTAJ: 'Smestaj',
  HRANA: 'Hrana',
  ULAZNICE: 'Ulaznice',
  KUPOVINA: 'Kupovina',
  OSTALO: 'Ostalo',
};

export function createExpense(data = {}) {
  return {
    id: data.id ?? null,
    tripPlanId: data.tripPlanId ?? null,
    name: data.name ?? '',
    category: data.category ?? ExpenseCategory.OSTALO,
    amount: data.amount ?? 0,
    date: data.date ?? '',
    description: data.description ?? '',
  };
}